commons-dlp/src/main/java/org/exoplatform/commons/dlp/service/impl/DlpPositiveItemServiceImpl.java

```diff
@@ -41,7 +48,35 @@ public void deleteDlpPositiveItem(Long itemId) {
     }
 
     @Override
-    public DlpPositiveItemEntity getDlpPositiveItemByReference(String itemReference) {
-        return dlpPositiveItemDAO.findDlpPositiveItemByReference(itemReference);
+    public DlpPositiveItem getDlpPositiveItemByReference(String itemReference) throws Exception {
+        DlpPositiveItemEntity dlpPositiveItemEntity = dlpPositiveItemDAO.findDlpPositiveItemByReference(itemReference);
+        return fillDlpPositiveItemFromEntity(dlpPositiveItemEntity);
+    }
+
+    @Override
+    public Long getDlpPositiveItemsCount() {
+        return dlpPositiveItemDAO.count();
+    }
+
+
+    private List<DlpPositiveItem> fillDlpPositiveItemsFromEntities(List<DlpPositiveItemEntity> dlpPositiveItemEntities) throws Exception {
+        List<DlpPositiveItem> dlpPositiveItems = new ArrayList<>();
+        for (DlpPositiveItemEntity dlpPositiveItemEntity : dlpPositiveItemEntities) {
+            DlpPositiveItem dlpPositiveItem = fillDlpPositiveItemFromEntity(dlpPositiveItemEntity);
+            dlpPositiveItems.add(dlpPositiveItem);
+        }
+        return dlpPositiveItems;
+    }
+
+    private DlpPositiveItem fillDlpPositiveItemFromEntity(DlpPositiveItemEntity dlpPositiveItemEntity) throws Exception {
+        DlpPositiveItem dlpPositiveItem = new DlpPositiveItem();
+        dlpPositiveItem.setId(dlpPositiveItemEntity.getId());
+        dlpPositiveItem.setType(dlpPositiveItemEntity.getType());
+        dlpPositiveItem.setKeywords(dlpPositiveItemEntity.getKeywords());
+        dlpPositiveItem.setAuthor(dlpPositiveItemEntity.getAuthor());
+        dlpPositiveItem.setAuthorFullName(organizationService.getUserHandler().findUserByName(dlpPositiveItemEntity.getAuthor()).getDisplayName());
```

Comment:
I prefer that DlpService not depends of orgService
You should return username here, and at display, get userfullname with rest service

Reply:
Updated !